README.md

# readmission

readmission is an R data package containing a data frame, `readmission`,
giving hospital readmission data for patients with diabetes. Clinical
care data is from 130 U.S. hospitals in years 1999-2008. Each row
describes a patient “encounter” with variables on demographics,
medications, patient history, diagnostics, payment, and readmission.

## Installation

You can install the most recent release of readmission on CRAN with:

``` r
install.packages("readmission")
```

Install the development version of readmission from
[GitHub](https://github.com/) with:

``` r
# install.packages("pak")
pak::pak("simonpcouch/readmission")
```

## Example

``` r
library(tibble)
library(readmission)

readmission
#> # A tibble: 71,515 × 12
#>    readmitted race   sex   age   admission_source blood_glucose insurer duration
#>    <fct>      <fct>  <fct> <fct> <fct>            <fct>         <fct>      <dbl>
#>  1 Yes        Afric… Male  [60-… Referral         <NA>          <NA>           7
#>  2 No         Cauca… Fema… [50-… Emergency        Normal        Private        4
#>  3 Yes        Cauca… Fema… [70-… Referral         <NA>          Medica…        5
#>  4 No         Cauca… Fema… [80-… Referral         <NA>          Private        5
#>  5 No         Cauca… Fema… [70-… Referral         <NA>          <NA>           4
#>  6 No         Cauca… Male  [50-… Emergency        Very High     <NA>           2
#>  7 Yes        Afric… Fema… [70-… Referral         <NA>          Private        3
#>  8 No         Cauca… Fema… [20-… Emergency        <NA>          <NA>           1
#>  9 No         Cauca… Male  [60-… Other            <NA>          <NA>          12
#> 10 No         Cauca… Fema… [80-… Referral         <NA>          Medica…        1
#> # ℹ 71,505 more rows
#> # ℹ 4 more variables: n_previous_visits <dbl>, n_diagnoses <dbl>,
#> #   n_procedures <dbl>, n_medications <dbl>
```


NEWS.md
# readmission (development version)

* Added a `NEWS.md` file to track changes to the package.

# readmission 0.1.0

Initial CRAN release.


data-raw/readmission_raw.R
library(tidyverse)

readmission_raw <- read_csv("https://github.com/fairlearn/talks/raw/8e768c372afb03099757c8ba2c43cfc95a681598/2021_scipy_tutorial/data/diabetic_data.csv")

# in case of repo deletion:
write.csv(readmission_raw, file = "data-raw/readmission_raw.csv")

set.seed(1)

readmission <-
  readmission_raw %>%
  filter(
    # 3 / 100k, and the label seems to be an artifact of data collection
    # rather than identification.
    gender != "Unknown/Invalid",
    # keep only the first visit and possible readmission for a patient
    !duplicated(patient_nbr)
  ) %>%
  transmute(
    # move outcome to first position
    readmitted = case_when(readmitted == "<30" ~ "Yes", .default = "No"),
    readmitted = factor(readmitted, levels = c("Yes", "No")),
    race = case_when(
      race == "AfricanAmerican" ~ "African American",
      race == "?" ~ "Unknown",
      .default = race
    ),
    # the entries in this column more accurately correspond to sex descriptors.
    # generally, though, more important to reason thoughtfully about how this
    # data was collected/coded (e.g. note no intersex patients).
    sex = gender,
    age = factor(age),
    admission_source = case_when(
      admission_source_id %in% c(1, 2, 3) ~ "Referral",
      admission_source_id == 7 ~ "Emergency",
      .default = "Other"
    ),
    blood_glucose = case_when(
      A1Cresult == "Norm" ~ "Normal",
      A1Cresult == ">7" ~ "High",
      A1Cresult == ">8" ~ "Very High",
      A1Cresult == "Norm" ~ NA_character_
    ),
    blood_glucose = factor(
      blood_glucose,
      levels = c("Normal", "High", "Very High")
    ),
    insurer = case_when(
      payer_code == "MC" ~ "Medicare",
      payer_code == "MD" ~ "Medicaid",
      payer_code == "SP" ~ "Self-Pay",
      payer_code == "?" ~ NA_character_,
      .default = "Private"
    ),
    duration = time_in_hospital,
    n_previous_visits = number_emergency + number_inpatient + number_outpatient,
    n_diagnoses = number_diagnoses,
    n_procedures = num_procedures,
    n_medications = num_medications
  ) %>%
  mutate(across(where(is.character), as.factor)) %>%
  slice_sample(prop = 1)

usethis::use_data(readmission, overwrite = TRUE)


R/readmission.R
#' Hospital Readmission Data for Patients with Diabetes
#'
#' Clinical care data from 130 U.S. hospitals in years 1999-2008.
#' Each row describes an "encounter" with a patient with diabetes, including
#' variables on demographics, medications, patient history, diagnostics,
#' payment, and readmission.
#'
#' @format
#' A data frame with 71,515 rows and 12 columns:
#' \describe{
#'   \item{readmitted}{Whether the patient was readmitted within the 30 days
#'   following discharge. A factor with levels `"Yes"` and `"No"`.}
#'   \item{race}{Reported race of the patient. Source data does not document
#'   data collection strategy. A factor with levels `"African American"`,
#'   `"Asian"`, `"Caucasian"`, `"Hispanic"`, `"Other"`, and `"Unknown"`.}
#'   \item{sex}{Reported sex of the patient. Source data does not document
#'   data collection strategy. A factor with levels `"Female"` and `"Male"`.}
#'   \item{age}{Age range for the patient, binned in 10-year intervals. A factor
#'   with levels `"[0-10)"`, `"[10-20)"`, `"[20-30)"`, `"[30-40)"`, `"[40-50)"`,
#'   `"[50-60)"`, `"[60-70)"`, `"[70-80)"`, `"[80-90)"`, and `"[90-100)"`.}
#'   \item{admission_source}{Whether the patient was referred from a physician,
#'   admitted via the ER, or arrived via some other source. A factor with
#'   levels `"Emergency"`, `"Other"`, and `"Referral"`.}
#'   \item{blood_glucose}{Results from an A1C test, estimating the
#'   patient's average blood sugar over the past 2-3 months. Higher estimated
#'   average blood glucose levels are linked to diabetes complications. A factor
#'   with levels `"Normal"`, `"High"`, and `"Very High"`, and many missing values.}
#'   \item{insurer}{The health insurance provider (or lack thereof,
#'   via `"Self-Pay"`) for the patient. A factor with levels
#'   `"Medicaid"`, `"Medicare"`, `"Private"`, and `"Self-Pay"`, and many missing
#'   values.}
#'   \item{duration}{Number of days in the hospital between admission and
#'   discharge.}
#'   \item{n_previous_visits}{Number of emergency, inpatient, and outpatient
#'   visits in the year preceding the encounter.}
#'   \item{n_diagnoses}{"Number of diagnoses entered to the system" during
#'   the encounter.}
#'   \item{n_procedures}{"Number of procedures (other than lab tests)
#'   performed" during the encounter.}
#'   \item{n_medications}{"Number of distinct generic names administered"
#'   during the encounter.}
#' }
#'
#' @source
#' _Original source data from the following paper (CC BY 3.0):_
#'
#' Strack, B., DeShazo, J. P., Gennings, C., Olmo, J. L., Ventura, S., Cios,
#' K. J., & Clore, J. N. 2014. Impact of HbA1c measurement on hospital
#' readmission rates: analysis of 70,000 clinical database patient records.
#' BioMed research international, 781670. <doi:10.1155/2014/781670>.
#'
#' _Shared freely through the UCI Machine Learning Repository (CC BY 4.0):_
#'
#' Clore, J., Cios, K., DeShazo, J. P., and Strack, B. 2014. Diabetes 130-US
#' hospitals for years 1999-2008. UCI Machine Learning Repository.
#' <doi:10.24432/C5230J>.
#'
#' _Downloaded from resources shared by the Fairlearn team (MIT):_
#'
#' Weerts, H., Dudík M., Edgar, R., Jalali, A., Lutz, R., & Madaio, M. 2023.
#' Fairlearn: Assessing and Improving Fairness of AI Systems. Journal of
#' Machine Learning Research, 24(257):1-8.
#'
#' @examples
#'
#' str(readmission)
#'
#' head(readmission)
#'
"readmission"
